src/EditorFeatures/Core/Shared/Utilities/ForegroundThreadAffinitizedObject.cs

```diff
@@ -66,19 +66,14 @@ internal static ForegroundThreadData CurrentForegroundThreadData
             }
         }
 
-        internal ForegroundThreadData ForegroundThreadData
-        {
-            get { return _foregroundThreadData; }
-        }
-
         internal Thread ForegroundThread
```

Comment:
Should we change the name of the prop to have WhenCreated suffix as well to indicate the scary nature?